redish/libs/redish-backend/rest-api/src/lib/controllers/user.controller.ts

```diff
@@ -0,0 +1,133 @@
+import { UuidDto } from '../dtos/uuid.dto';
+import { CreateUserDto } from '../dtos/create-user.dto';
+import {
+  ApiBadRequestResponse,
+  ApiBearerAuth,
+  ApiCreatedResponse,
+  ApiInternalServerErrorResponse,
+  ApiOkResponse,
+  ApiTags,
+  ApiUnauthorizedResponse,
+} from '@nestjs/swagger';
+import { AuthGuard } from '../guards/auth.guard';
+import { Response } from 'express';
+import { RedishErrorDto } from '../dtos/redish-error.dto';
+import { RedishError, UpdateUserDto } from '@redish-shared/domain';
+import { AuthenticatedRequest } from '../types/authenticated-request.type';
+
+@ApiTags('user')
+@Controller('user')
+@Injectable()
+export class UserController {
+  constructor(private authenticationFacade: AuthenticationFacade) {}
+
+  @ApiOkResponse({ type: TokenDto })
+  @ApiBadRequestResponse({ type: RedishErrorDto })
+  @ApiInternalServerErrorResponse({ type: RedishErrorDto })
+  @Post('login')
+  public async authenticateUser(
+    @Res({ passthrough: true }) response: Response,
+    @Body() authenticateUserDto: AuthenticateUserDto
+  ): Promise<TokenDto | RedishErrorDto> {
+    const authResult = await this.authenticationFacade.authenticateUser(
+      authenticateUserDto
+    );
+    if (authResult.error) {
+      if (
+        authResult.error.code ===
+        RedishError.Infrastructure.Codes.DATABASE_ERROR
+      ) {
+        response.status(HttpStatus.INTERNAL_SERVER_ERROR).send();
+      } else {
+        response.status(HttpStatus.BAD_REQUEST).send();
+      }
+      return authResult.error;
+    } else {
+      if (authResult.result !== null) {
+        return authResult.result;
+      } else {
+        return RedishError.Infrastructure.databaseError();
+      }
+    }
+  }
+
+  @ApiCreatedResponse({ type: UuidDto })
+  @ApiBadRequestResponse({ type: RedishErrorDto })
+  @ApiInternalServerErrorResponse({ type: RedishErrorDto })
+  @Post()
+  async createUser(
+    @Res({ passthrough: true }) response: Response,
+    @Body() createUserDto: CreateUserDto
+  ): Promise<UuidDto | RedishErrorDto> {
+    const createUserResult = await this.authenticationFacade.createUser(
+      createUserDto
+    );
+    if (createUserResult.error) {
+      if (
+        createUserResult.error.code ===
+        RedishError.Infrastructure.Codes.DATABASE_ERROR
+      ) {
+        response.status(HttpStatus.INTERNAL_SERVER_ERROR).send();
+      } else {
+        response.status(HttpStatus.BAD_REQUEST).send();
+      }
+      return createUserResult.error;
+    } else {
+      return createUserResult.result!;
+    }
+  }
+
+  @ApiOkResponse({ type: UuidDto })
+  @ApiBadRequestResponse({ type: RedishErrorDto })
+  @ApiInternalServerErrorResponse({ type: RedishErrorDto })
+  @ApiUnauthorizedResponse({ type: RedishErrorDto })
+  @UseGuards(AuthGuard)
+  @Put()
+  async updateUser(
+    @Res({ passthrough: true }) response: Response,
+    @Req() request: AuthenticatedRequest,
+    @Body() updateUserDto: UpdateUserDto
```

Comment:
UpdateUserDto is missing in `../dtos/`. That is why in swagger you can't see the model and can't try the request